compositor_chromium/src/v8/function.rs

```diff
@@ -0,0 +1,168 @@
+use std::{fmt::Display, os::raw::c_int};
+
+use crate::{
+    cef_ref::{CefRefData, CefStruct},
+    cef_string::CefString,
+    validated::{Validated, ValidatedError},
+};
+
+use super::{
+    value::{V8Value, V8ValueError},
+    V8ContextEntered, V8Object,
+};
+
+pub struct V8Function(pub(super) Validated<chromium_sys::cef_v8value_t>);
+
+impl V8Function {
+    pub fn new<F>(name: &str, func: F) -> Self
+    where
+        F: Fn(&[V8Value]) -> Result<V8Value, NativeFunctionError>,
+    {
+        let name = CefString::new_raw(name);
+        let handler = CefRefData::new_ptr(V8Handler(func));
+        let inner = unsafe { chromium_sys::cef_v8value_create_function(&name, handler) };
+
+        Self(Validated::new(inner))
+    }
+
+    pub fn call(
+        &self,
+        args: &[V8Value],
+        ctx_entered: &V8ContextEntered,
+    ) -> Result<V8Value, V8FunctionError> {
+        self.inner_call(None, args, ctx_entered)
+    }
```

Comment:
If F panics, the unwinding panic will cross into C++ land, which is a big no-no. 
read [this](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html) and catch the panic.